pkgs/build-support/node/fetch-yarn-deps/default.nix

```
@@ -131,23 +134,68 @@ in
             dontUnpack = src == null;
             dontInstall = true;
 
-            nativeBuildInputs = [
-              prefetch-yarn-deps
-              cacert
-            ];
+            nativeBuildInputs =
+              [ cacert ]
+              ++ (
+                if (yarnVersion == 1) then
+                  [ prefetch-yarn-deps ]
+                else if (yarnVersion == 3) then
+                  [ yarn-berry3 ]
+                else if (yarnVersion == 4) then
+                  [ yarn-berry4 ]
+                else
+                  [ ]
+              );
```

Readability improvement:

```suggestion
            nativeBuildInputs = [
              ({
                1 = prefetch-yarn-deps;
                3 = yarn-berry3;
                4 = yarn-berry4;
              }.${yarnVersion})
              cacert
            ];
```

Also, it will naturally error when an unsupported version will be given.

---

thanks!